toolchain/check/handle_function.cpp

```diff
@@ -194,9 +194,17 @@ static auto BuildFunctionDecl(Context& context,
   DiagnoseModifiers(context, introducer, is_definition, parent_scope_inst_id,
                     parent_scope_inst);
   bool is_extern = introducer.modifier_set.HasAnyOf(KeywordModifierSet::Extern);
-  if (introducer.modifier_set.HasAnyOf(KeywordModifierSet::Method)) {
-    context.TODO(introducer.modifier_node_id(ModifierOrder::Decl),
-                 "method modifier");
+  SemIR::FunctionFields::VirtualModifier virtual_modifier =
+      SemIR::FunctionFields::VirtualModifier::None;
+
+  if (introducer.modifier_set.HasAnyOf(KeywordModifierSet::Virtual)) {
+    virtual_modifier = SemIR::FunctionFields::VirtualModifier::Virtual;
+  }
+  if (introducer.modifier_set.HasAnyOf(KeywordModifierSet::Abstract)) {
+    virtual_modifier = SemIR::FunctionFields::VirtualModifier::Abstract;
+  }
+  if (introducer.modifier_set.HasAnyOf(KeywordModifierSet::Impl)) {
+    virtual_modifier = SemIR::FunctionFields::VirtualModifier::Impl;
   }
```

Comment:
Note, not suggesting a change here, but thinking about this...

Compare what you're doing here with the `class` code:
https://github.com/carbon-language/carbon-lang/blob/trunk/toolchain/check/handle_class.cpp#L203-L208

Maybe we should have a helper or something on `KeywordModifierSet` to help switch modifiers to an appropriate enum? It feels like something that we could maybe do with a variadic template...

Reply:
Oh, yeah - I can look into that.

Reply:
Sent https://github.com/carbon-language/carbon-lang/pull/4290

Aside: The `handle_class.cpp` case uses a non-scoped enum (`enum`, not `enum class`) and references the entities via the derived type (`Class::Enumerator` rather than `ClassFields::Enumerator`) - I guess the latter's clearly an improvement over the code I've written here for the function case, but should we standardize on the enum or enum class here? In this particular case an unscoped enum seems OK?